### DISCUSSION TITLE
Dynamic Weather Icon

### DISCUSSION THREAD

**Discussion by Jab2870** [elax46/custom-brand-icons#688] [Ideas]
Not sure if this is even possible, but would it be possible to have an icon that was something like weather-current? Or weather-{weather-entity}-current? The idea would be to have a single icon that can be used on pages or dashboards that reacts to the current weather.

The result would be having this if it were partly cloudy
![image](https://github.com/elax46/custom-brand-icons/assets/1909202/3f174243-2e86-4167-b558-f64a29a3ad59)

or this if it were snowing
![image](https://github.com/elax46/custom-brand-icons/assets/1909202/f632e0e8-430e-46c1-8265-8cfed554932e)

And so on.

**Comment by rchiileea** [comment]
it would be possible, but you would have to yaml code it yourself to react like this

**Reply by Jab2870** [reply]
My initial thought had been to use templates, something like:

```yaml
title: Weather
path: weather
icon: >
  {% if states('weather.forecast_home') == "cloudy" %}
  mdi:weather-cloudy
  {% else %}
  mdi:weather-sunny
  {% endif %}
cards:
  ...
```

However, templates don't work here. Is this what you were referring to?

**Reply by rchiileea** [reply]
Yes there is no way for us to influence what icon shows when that’s on the core side for menu tabs, only way is to create your own header menu

**Reply by Jab2870** [reply]
Could you point me towards some documentation for that?

**Comment by rchiileea** [comment]
![image](https://github.com/elax46/custom-brand-icons/assets/87599807/4bb950cb-fb4f-4a69-97e4-2463f6516acb)
so you would have to add a header under the menu, for example my header has good afternoon, then info under it sliding acrross from right to left stops for a second then more info (i have for example which windows and doors are open show up (conditional), if car is bellow 50% to say it needs a charge etc. also custom icons to the right, but this is done through a complete yaml only dashboard called hki
